src/ontodia/diagram/paperArea.tsx

```diff
@@ -543,80 +571,96 @@ export class PaperArea extends React.Component<PaperAreaProps, State> {
         }
     }
 
-    centerTo(paperPosition?: { x: number; y: number }) {
-        const {paperWidth, paperHeight, scale, originX, originY, paddingX, paddingY} = this.state;
+    /**
+     * animatable operation (return await promise)
+     */
+    centerTo(paperPosition?: { x: number; y: number }, animate?: ViewportAnimationOptions): Promise<void> {
+        const {paperWidth, paperHeight} = this.state;
         const paperCenter = paperPosition || {x: paperWidth / 2, y: paperHeight / 2};
-        const clientCenterX = (paperCenter.x + originX) * scale;
-        const clientCenterY = (paperCenter.y + originY) * scale;
-        const {clientWidth, clientHeight} = this.area;
-        this.area.scrollLeft = clientCenterX - clientWidth / 2 + paddingX;
-        this.area.scrollTop = clientCenterY - clientHeight / 2 + paddingY;
+        const viewportState: Partial<ViewportState> = {
+            center: paperCenter,
+        };
+        return this.setViewportState(viewportState, animate);
     }
 
-    centerContent() {
+    /**
+     * animatable operation (return await promise)
+     */
+    centerContent(animate?: ViewportAnimationOptions): Promise<void> {
         const bbox = this.getContentFittingBox();
-        this.centerTo({
+        return this.centerTo({
             x: bbox.x + bbox.width / 2,
             y: bbox.y + bbox.height / 2,
-        });
+        }, animate);
     }
 
     getScale() {
         return this.state.scale;
     }
 
-    setScale(value: number, options: ScaleOptions = {}) {
+    /**
+     * animatable operation (return await promise)
+     */
+    setScale(value: number, options: ScaleOptions = {}, animate?: ViewportAnimationOptions): Promise<void> {
         let scale = value;
 
         const {min, max} = this.zoomOptions;
         scale = Math.max(scale, min);
         scale = Math.min(scale, max);
 
-        const center = this.clientToPaperCoords(
+        const paperCenter = this.clientToPaperCoords(
```

Comment:
You may want to move this assignment to `if-then`  branch (it only used when there is a `pivot`).